### ISSUE TITLE
v0.0.0.69 GUI contract creation doesn't work

### ISSUE THREAD

**Issue by walkjivefly** [Crowndev/crown#188]
Attempting to create a new contract from the Assets page has a number of issues:
1. The address selector (I assume that's what it is since it's unlabelled) is greyed out and does nothing.
2. If an address is entered in the unlabelled box below it, on hitting Create a popup is displayed saying "Private key for address <whatever> is unknown"
![Screenshot_2022-11-06_17-19-15](https://user-images.githubusercontent.com/17745270/200182144-2e3f965a-3aa7-4da5-9ede-881a49b55b55.png)

**Comment by BlockMechanic** [comment]
Yes the selector is disabled for now, some of that code is finicky and needs attention. That is accurate, do you have the private key ? Remember last time you struggled because the wallet was locked.

**Comment by walkjivefly** [comment]
Oh, a locked wallet is a distinct possibility. I'm still trying to get the wallet synced following the crash so can't confirm it yet. But if that is the case, I suggest a more accurate/descriptive error message would be a good thing (TM).

**Comment by walkjivefly** [comment]
This works OK with an unencrypted (hence, unlocked) wallet (see contract C3). 

With an encrypted, locked wallet v0.0.0.72 asks for the passphrase and then crashes with SIGSEGV. debug.log shows
```
2022-11-16T21:42:40Z GUI: "C4"   "C4"   "tCRWN6ngPvTBiGrDzsFSLYGMt19qEy57Qh1Mt"   "ibm.com"   "ibm.com"   "yada yada yada"   ""
2022-11-16T21:42:49Z GUI: Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.
```
and gdb says
```
Core was generated by `crown-qt'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x000056310a1e5de8 in QEventLoop::exit(int) ()
[Current thread is 1 (Thread 0x7f46ae320800 (LWP 1532))]
(gdb) bt
#0  0x000056310a1e5de8 in QEventLoop::exit(int) ()
#1  0x000056310a62eff2 in QDialog::setVisible(bool) ()
#2  0x000056310a62c951 in QDialog::~QDialog() ()
#3  0x00005631093234bf in AssetManagerPage::on_CreateNewContract_clicked() [clone .cold] ()
#4  0x00005631094f25dd in AssetManagerPage::qt_metacall(QMetaObject::Call, int, void**) ()
#5  0x000056310a213568 in void doActivate<false>(QObject*, int, void**) ()
#6  0x000056310a5541b5 in QAbstractButtonPrivate::emitClicked() ()
#7  0x000056310a555f4b in QAbstractButtonPrivate::click() ()
#8  0x000056310a556167 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) ()
#9  0x000056310a4d51a8 in QWidget::event(QEvent*) ()
#10 0x000056310a49c533 in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    ()
#11 0x000056310a4a4ed2 in QApplication::notify(QObject*, QEvent*) ()
#12 0x000056310a1e8288 in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) ()
#13 0x000056310a4a38b6 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) ()
#14 0x000056310a4ee375 in QWidgetWindow::handleMouseEvent(QMouseEvent*) ()
#15 0x000056310a4f15b5 in QWidgetWindow::event(QEvent*) ()
#16 0x000056310a49c533 in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    ()
#17 0x000056310a1e8288 in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) ()
#18 0x0000563109ca56bf in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) ()
#19 0x0000563109f2fa9c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
#20 0x000056310a38ab97 in QXcbUnixEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
#21 0x000056310a1e6486 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
#22 0x000056310a1ef988 in QCoreApplication::exec() ()
#23 0x000056310943431a in GuiMain(int, char**) ()
#24 0x00007f46ae450d90 in __libc_start_call_main (
    main=main@entry=0x563109415e60 <main>, argc=argc@entry=1, 
    argv=argv@entry=0x7ffd7be29ce8)
    at ../sysdeps/nptl/libc_start_call_main.h:58
#25 0x00007f46ae450e40 in __libc_start_main_impl (main=0x563109415e60 <main>, 
    argc=1, argv=0x7ffd7be29ce8, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7ffd7be29cd8)
    at ../csu/libc-start.c:392
#26 0x000056310942dd25 in _start ()
(gdb) info threads
  Id   Target Id                        Frame 
* 1    Thread 0x7f46ae320800 (LWP 1532) 0x000056310a1e5de8 in QEventLoop::exit(int) ()
  2    Thread 0x7f46a6ffd640 (LWP 1536) 0x00007f46ae53fe7e in __ppoll (
    fds=0x7f469c01bd78, nfds=6, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  3    Thread 0x7f46a67fc640 (LWP 1537) 0x00007f46ae53fe7e in __ppoll (
    fds=0x7f469201e5f8, nfds=1, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  4    Thread 0x7f468bfff640 (LWP 1544) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x7f469004d5f0) at ./nptl/futex-internal.c:57
  5    Thread 0x7f46a4f64640 (LWP 1538) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x56310b6cda90 <scriptcheckqueue+80>)
    at ./nptl/futex-internal.c:57
  6    Thread 0x7f46967fd640 (LWP 1540) 0x00007f46ae54cfde in epoll_wait (
    epfd=20, events=0x7f469006ff10, maxevents=32, timeout=-1)
    at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
  7    Thread 0x7f46777fe640 (LWP 1547) 0x00007f46ae54cfde in epoll_wait (
    epfd=50, events=0x7f466d513e30, maxevents=32, timeout=985261)
    at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
  8    Thread 0x7f4695ffc640 (LWP 1541) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x7f469004d5f4) at ./nptl/futex-internal.c:57
  9    Thread 0x7f46957fb640 (LWP 1542) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x7f469004d5f0) at ./nptl/futex-internal.c:57
  10   Thread 0x7f4666ffd640 (LWP 1555) 0x00007f46ae53fe7e in __ppoll (
    fds=0x7f467c001038, nfds=1, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  11   Thread 0x7f4694ffa640 (LWP 1543) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x7f469004d5f4) at ./nptl/futex-internal.c:57
  12   Thread 0x7f46a7fff640 (LWP 1618) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x7f46a7ffec70, op=137, expected=0, 
    futex_word=0x56310b9aff20) at ./nptl/futex-internal.c:57
  13   Thread 0x7f46a77fe640 (LWP 1619) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x7f46a77fdc70, op=137, expected=0, 
    futex_word=0x56310b9afcf4) at ./nptl/futex-internal.c:57
  14   Thread 0x7f46757fa640 (LWP 1551) __futex_abstimed_wait_common64 (
    private=191692248, cancel=true, abstime=0x7f46757f9b00, op=137, 
    expected=0, futex_word=0x7f46901588b0) at ./nptl/futex-internal.c:57
  15   Thread 0x7f4696ffe640 (LWP 1539) __futex_abstimed_wait_common64 (
    private=-1510251008, cancel=true, abstime=0x7f4696ffdc40, op=393, 
    expected=0, futex_word=0x7f4690001578) at ./nptl/futex-internal.c:57
  16   Thread 0x7f4676ffd640 (LWP 1548) 0x00007f46ae53fd7f in __GI___poll (
    fds=0x56310bea2fb0, nfds=11, timeout=50)
    at ../sysdeps/unix/sysv/linux/poll.c:29
--Type <RET> for more, q to quit, c to continue without paging--
  17   Thread 0x7f46ad341640 (LWP 1533) 0x00007f46ae53fd7f in __GI___poll (
    fds=0x7f46ad340c08, nfds=1, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
  18   Thread 0x7f4675ffb640 (LWP 1550) __futex_abstimed_wait_common64 (
    private=232008040, cancel=true, abstime=0x7f4675ffab30, op=137, 
    expected=0, futex_word=0x7f46901588b0) at ./nptl/futex-internal.c:57
  19   Thread 0x7f4667fff640 (LWP 1553) __futex_abstimed_wait_common64 (
    private=1744824920, cancel=true, abstime=0x7f4667ffec40, op=137, 
    expected=0, futex_word=0x7f4690158854) at ./nptl/futex-internal.c:57
  20   Thread 0x7f46677fe640 (LWP 1554) 0x00007f46ae53fe7e in __ppoll (
    fds=0x7f465c000e58, nfds=1, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  21   Thread 0x7f4674ff9640 (LWP 1552) __futex_abstimed_wait_common64 (
    private=1797285236, cancel=true, abstime=0x7f4674ff84f0, op=137, 
    expected=0, futex_word=0x7f46901588b0) at ./nptl/futex-internal.c:57
  22   Thread 0x7f46667fc640 (LWP 1557) 0x00007f46ae53fe7e in __ppoll (
    fds=0x7f4678001068, nfds=1, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  23   Thread 0x7f4665ffb640 (LWP 1558) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x56310b6d783c <leveldb::Env::Default()::env_container+92>)
    at ./nptl/futex-internal.c:57
```
debug.log and core [here](https://storage.crownplatform.com/s/jbmE9ZR5Q3Hf6z7)

With an encrypted, unlocked wallet it still crashes in the same way, debug.log shows
```
2022-11-16T22:05:21Z GUI: "C4"   "C4"   "tCRWN6ngPvTBiGrDzsFSLYGMt19qEy57Qh1Mt"   "oracle.com"   "oracle.com"   "crash or not"   ""
2022-11-16T22:05:22Z GUI: Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.
```
and gdb shows similar backtrace and threads as before
```
Core was generated by `crown-qt'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00005653b27c7de8 in QEventLoop::exit(int) ()
[Current thread is 1 (Thread 0x7f54469e9800 (LWP 1640))]
(gdb) bt
#0  0x00005653b27c7de8 in QEventLoop::exit(int) ()
#1  0x00005653b2c10ff2 in QDialog::setVisible(bool) ()
#2  0x00005653b2c0e951 in QDialog::~QDialog() ()
#3  0x00005653b19054bf in AssetManagerPage::on_CreateNewContract_clicked() [clone .cold] ()
#4  0x00005653b1ad45dd in AssetManagerPage::qt_metacall(QMetaObject::Call, int, void**) ()
#5  0x00005653b27f5568 in void doActivate<false>(QObject*, int, void**) ()
#6  0x00005653b2b361b5 in QAbstractButtonPrivate::emitClicked() ()
#7  0x00005653b2b37f4b in QAbstractButtonPrivate::click() ()
#8  0x00005653b2b38167 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) ()
#9  0x00005653b2ab71a8 in QWidget::event(QEvent*) ()
#10 0x00005653b2a7e533 in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    ()
#11 0x00005653b2a86ed2 in QApplication::notify(QObject*, QEvent*) ()
#12 0x00005653b27ca288 in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) ()
#13 0x00005653b2a858b6 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) ()
#14 0x00005653b2ad0375 in QWidgetWindow::handleMouseEvent(QMouseEvent*) ()
#15 0x00005653b2ad35b5 in QWidgetWindow::event(QEvent*) ()
#16 0x00005653b2a7e533 in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    ()
#17 0x00005653b27ca288 in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) ()
#18 0x00005653b22876bf in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) ()
#19 0x00005653b2511a9c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
#20 0x00005653b296cb97 in QXcbUnixEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
#21 0x00005653b27c8486 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) ()
#22 0x00005653b27d1988 in QCoreApplication::exec() ()
#23 0x00005653b1a1631a in GuiMain(int, char**) ()
#24 0x00007f5446b19d90 in __libc_start_call_main (
    main=main@entry=0x5653b19f7e60 <main>, argc=argc@entry=1, 
    argv=argv@entry=0x7ffd8ed61de8)
    at ../sysdeps/nptl/libc_start_call_main.h:58
#25 0x00007f5446b19e40 in __libc_start_main_impl (main=0x5653b19f7e60 <main>, 
    argc=1, argv=0x7ffd8ed61de8, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7ffd8ed61dd8)
    at ../csu/libc-start.c:392
#26 0x00005653b1a0fd25 in _start ()
(gdb) info threads
  Id   Target Id                        Frame 
* 1    Thread 0x7f54469e9800 (LWP 1640) 0x00005653b27c7de8 in QEventLoop::exit(int) ()
  2    Thread 0x7f5445a0a640 (LWP 1641) 0x00007f5446c08d7f in __GI___poll (
    fds=0x7f5445a09c08, nfds=1, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
  3    Thread 0x7f543f7fe640 (LWP 1644) 0x00007f5446c08e7e in __ppoll (
    fds=0x7f5434019648, nfds=6, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  4    Thread 0x7f543eebf640 (LWP 1645) 0x00007f5446c08e7e in __ppoll (
    fds=0x7f53f4064518, nfds=1, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  5    Thread 0x7f542ffff640 (LWP 1646) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x5653b3cafa90 <scriptcheckqueue+80>)
    at ./nptl/futex-internal.c:57
  6    Thread 0x7f542effd640 (LWP 1648) 0x00007f5446c15fde in epoll_wait (
    epfd=20, events=0x7f54280709b0, maxevents=32, timeout=2100000)
    at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
  7    Thread 0x7f542e7fc640 (LWP 1649) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x7f542805de00) at ./nptl/futex-internal.c:57
  8    Thread 0x7f542dffb640 (LWP 1650) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x7f542805de00) at ./nptl/futex-internal.c:57
  9    Thread 0x7f542d7fa640 (LWP 1651) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x7f542805de00) at ./nptl/futex-internal.c:57
  10   Thread 0x7f53fb7fe640 (LWP 1661) __futex_abstimed_wait_common64 (
    private=948113832, cancel=true, abstime=0x7f53fb7fdb30, op=137, 
    expected=0, futex_word=0x7f5428160d20) at ./nptl/futex-internal.c:57
  11   Thread 0x7f53faffd640 (LWP 1662) __futex_abstimed_wait_common64 (
    private=-1278534184, cancel=true, abstime=0x7f53faffcb00, op=137, 
    expected=0, futex_word=0x7f5428160d20) at ./nptl/futex-internal.c:57
  12   Thread 0x7f53f8ff9640 (LWP 1666) 0x00007f5446c08e7e in __ppoll (
    fds=0x7f5410001038, nfds=1, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  13   Thread 0x7f53effff640 (LWP 1668) 0x00007f5446c08e7e in __ppoll (
    fds=0x7f540c001068, nfds=1, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  14   Thread 0x7f540a18c640 (LWP 1677) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x7f540a18bc70, op=137, expected=0, 
    futex_word=0x5653b53cdf60) at ./nptl/futex-internal.c:57
  15   Thread 0x7f542cff9640 (LWP 1652) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x7f542805de00) at ./nptl/futex-internal.c:57
  16   Thread 0x7f543ffff640 (LWP 1658) 0x00007f5446c15fde in epoll_wait (
    epfd=7, events=0x7f53f4098d00, maxevents=32, timeout=86497)
    at ../sysdeps/unix/sysv/linux/epoll_wait.c:30
--Type <RET> for more, q to quit, c to continue without paging--
  17   Thread 0x7f53ef7fe640 (LWP 1669) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x0, op=393, expected=0, 
    futex_word=0x5653b3cb983c <leveldb::Env::Default()::env_container+92>)
    at ./nptl/futex-internal.c:57
  18   Thread 0x7f53ee5c5640 (LWP 1676) __futex_abstimed_wait_common64 (
    private=0, cancel=true, abstime=0x7f53ee5c4c70, op=137, expected=0, 
    futex_word=0x5653b53cdd30) at ./nptl/futex-internal.c:57
  19   Thread 0x7f542f7fe640 (LWP 1647) __futex_abstimed_wait_common64 (
    private=868797440, cancel=true, abstime=0x7f542f7fdc40, op=393, 
    expected=0, futex_word=0x7f5428001a58) at ./nptl/futex-internal.c:57
  20   Thread 0x7f53f97fa640 (LWP 1665) 0x00007f5446c08e7e in __ppoll (
    fds=0x7f5404001488, nfds=1, timeout=<optimized out>, sigmask=0x0)
    at ../sysdeps/unix/sysv/linux/ppoll.c:42
  21   Thread 0x7f53f9ffb640 (LWP 1664) __futex_abstimed_wait_common64 (
    private=-100685224, cancel=true, abstime=0x7f53f9ffac40, op=137, 
    expected=0, futex_word=0x7f5428160cc0) at ./nptl/futex-internal.c:57
  22   Thread 0x7f54089c9640 (LWP 1659) 0x00007f5446c08d7f in __GI___poll (
    fds=0x7f53f00016a0, nfds=11, timeout=50)
    at ../sysdeps/unix/sysv/linux/poll.c:29
  23   Thread 0x7f53fa7fc640 (LWP 1663) __futex_abstimed_wait_common64 (
    private=1797285236, cancel=true, abstime=0x7f53fa7fb4f0, op=137, 
    expected=0, futex_word=0x7f5428160d20) at ./nptl/futex-internal.c:57
```
debug.log and core [here](https://storage.crownplatform.com/s/7M2FWo8ZAMAeDtc)

**Comment by walkjivefly** [comment]
Confirmed fixed in v0.0.0.75
